app/controllers/test_reports_controller.rb

```diff
@@ -2,29 +2,34 @@
 
 class TestReportsController < ApplicationController
   before_action :check_round, only: %i[show update]
-
   def index
     per_page = 30
-    @jobs = Job.all
-               .page(params[:page])
-               .per(per_page)
-               .order('jobs.id DESC')
-    job_ids = Job.select('id').page(params[:page]).per(per_page).order('jobs.id DESC').to_a
-    @joined_jobs = Job.join_with_suites
-                      .select_group_concat_suites
-                      .group('jobs.id')
-                      .where(id: job_ids)
-                      .order('jobs.id DESC')
+    all_roots = Job.all.where("command_and_option not like '%rerun%'").order('jobs.id DESC')
+    @page = Kaminari.paginate_array(all_roots.to_a, total_count: all_roots.length).page(params[:page]).per(per_page)
+    start_num = params[:page].nil? || params[:page] == 1 ? 0 : per_page * (params[:page].to_i - 1)
+    root_jobs = Job.join_with_suites(all_roots.to_a.map(&:id)[start_num...start_num + per_page])
+    children_jobs = Job.join_with_suites(Job.select('id').where("command_and_option like '%rerun%'").page(params[:page]).per(per_page * 3).order('jobs.id DESC').to_a)
```

Comment:
(Same) It might be a good idea to move to job model as this line depends on job tables in in database.